scripts/create_cu_vm.py

```diff
@@ -0,0 +1,113 @@
+import subprocess
+import pytest
+
+from utils.conf import cfme_data, credentials
+from utils.log import logger
+from utils.path import scripts_path
+from utils.providers import get_mgmt
+from utils.ssh import SSHClient
+
+
+def make_ssh_client(ip, sshname, sshpass):
+    connect_kwargs = {
+        'username': sshname,
+        'password': sshpass,
+        'hostname': ip
+    }
+    logger.info('In make_ssh_client')
+    return SSHClient(**connect_kwargs)
+
+
+def config_cu_vm(ssh_client):
+    """Downloads cu scripts, installs iperf using ssh_client
+
+    Args:
+        ssh_client: :py:class:`utils.ssh.SSHClient` instance
+    """
+    # TO DO: Get URL from cfme_data.yaml
+    url = 'http://qeblade5.rhq.lab.eng.bos.redhat.com/isos/debug'
+
+    logger.info('In config_vm')
+
+    exit_status, output = ssh_client.run_command(
+        "cd /etc/init.d; wget {}/cu-disk-script.sh".format(url))
+    if exit_status != 0:
+        logger.info('CU: There was an error downloading cu-disk-script.sh file')
+        sys.exit(127)
+    exit_status, output = ssh_client.run_command(
+        "chmod +x /etc/init.d/cu-disk-script.sh")
+    if exit_status != 0:
+        logger.info('There was an error running chmod')
+        sys.exit(127)
+    exit_status, output = ssh_client.run_command(
+        "cd /tmp; wget {}/crontab.in".format(url))
+    if exit_status != 0:
+        logger.info('CU: There was an error downloading crontab.in')
+        sys.exit(127)
+    exit_status, output = ssh_client.run_command(
+        "crontab /tmp/crontab.in")
+    if exit_status != 0:
+        logger.info('CU: There was an error installing crontab:')
+        sys.exit(127)
+    exit_status, output = ssh_client.run_command(
+        "yum install -y iperf")
+    if exit_status != 0:
+        logger.info('CU: There was an error installing iperf')
+        sys.exit(127)
+
+
+def reboot_vm(ssh_client):
+    """Reboot VM
+
+    Args:
+        ssh_client: :py:class:`utils.ssh.SSHClient` instance
+    """
+    logger.info('In reboot_vm')
+    command1 = 'reboot &'
+    exit_status, output = ssh_client.run_command(command1)
+    if exit_status != 0:
+        logger.info('CU: There was an error running reboot')
+        sys.exit(127)
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument('--provider', help='provider key in cfme_data')
+    parser.add_argument('--template', help='the name of the template to clone')
+    parser.add_argument('--vm_name', help='the name of the VM to create')
+    parser.add_argument('--purpose', help='specify capandu for a CU VM')
+
+    script = scripts_path.join('clone_template.py')
+    pytest.set_trace()
+    args = parser.parse_args()
+
+    params = ' '.join([str(script), '--provider', args.provider, '--vm_name', args.vm_name,
+        '--template', args.template, '--purpose', args.purpose])
+
+    subprocess.call(params, shell=True)
```

Comment:
Actually, I think you do not need to use deploy_template *if you put the code for purpose handling in the deploy_template function*.

```python
from utils.virtual_machines import deploy_template

deploy_template(args.provider, args.vm_name, template_name=args.template, purpose=args.purpose)
```

By not using subprocess we can see the error if it happens.